### PR TITLE
Adding a CNS with LES MMS.

### PR DESCRIPTION
**Pull request by marchdf**
This uses the automatic differentiation to create an MMS for the compressible Navier-Stokes equations with an constant Smagorinsky LES models. There is also an associated test for this solution. Please let me know if there is anything I should change/comment/etc. Thank you!

Supersedes #47

### CODE REVIEW

File: AUTHORS [hunk 1]
```diff
@@ -21,6 +21,7 @@ Marco Panesi         (Radiation)
 Gabriel Terejanu     (SMASA)
 Rhys Ulerich
 Jeremy Melvin        (Incomp NS with Boussinesq Buoyancy)
+Marc Henry de Frahan (Compressible NS with LES models)
```
Comment (nicholasmalaya) [inline AUTHORS]: 👍

File: CHANGES [hunk 2]
```diff
@@ -2,6 +2,7 @@ Version 0.51.0 (IN PROGRESS)
 
   * Updates and bugfixes coming!
   * Added Homogeneous Incompressible Navier Stokes with Boussinesq Buoyancy term
+  * Added compressible Navier-Stokes with LES models
```
Comment (nicholasmalaya) [inline CHANGES]: Can you specify the LES model? E.g., Smagorinsky, etc.

Comment (marchdf) [inline CHANGES]: Sure! See f44e33d

**Comment by marchdf** [comment]
Thanks for reviewing! In response to your comments: 
1. I would be happy to document the precise model. I will add tex/pdf files in a folder  `analytical/navier_stokes/navier_stokes_comp_steady_les/latex` if that's ok. 
2. I have used it to verify a codebase (a finite volume combustion code called PeleC here at NREL). I can put some graphs from that study in the documentation if that's appropriate.

Eventually I think I will want to use this framework with other LES models (dynamic Smagorinsky, etc). We shall see how that pans out.

**Comment by nicholasmalaya** [comment]
> I would be happy to document the precise model. I will add tex/pdf files in a folder analytical/navier_stokes/navier_stokes_comp_steady_les/latex if that's ok.

That would be really great. 

> I have used it to verify a codebase (a finite volume combustion code called PeleC here at NREL). I can put some graphs from that study in the documentation if that's appropriate.

Anything you have ready-made and good to plop in would be useful. 

Thanks. I'm happy to merge this, unless @roystgnr has any comments/reservations.

**Comment by roystgnr** [comment]
Yeah; I'd say commit whatever documentation you can toss in quickly, then we're good to merge.

File: src/masa_internal.h [hunk 3]
```diff
@@ -2070,6 +2070,79 @@ class ad_cns_3d_crossterms : public manufactured_solution<Scalar>
 
 
 // ------------------------------------------------------
+// --------------- ad_cns_3d_les
+// ------------------------------------------------------
```
Comment (nicholasmalaya) [inline src/masa_internal.h]: Would you add a reference to the location of the source documentation here? Might be as simple as,

"// Source Documentation available at: analytical/navier_stokes/navier_stokes_comp_steady_les/latex"

After that I'll merge.

Comment (marchdf) [inline src/masa_internal.h]: Yep, just pushed it.

**Comment by nicholasmalaya** [comment]
Thank you for your contribution to MASA!

**Comment by marchdf** [comment]
Pleasure was all mine. Loved working with it and look forward to keep doing so.

**Pull request merged**